src/Quartz/Impl/JobType.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+
+using Quartz.Util;
+
+namespace Quartz.Impl;
+
+/// <summary>
+/// Store the Job Type and backing field name for serialization
+/// </summary>
+public class JobType
```

Comment:
this should be sealed, I think also allowing implicit conversion for `Type` might be a good thing?